server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ChunkOutputStream.java

```diff
@@ -61,14 +62,16 @@
        this.completionSink = Objects.requireNonNull(completionSink);
        Objects.requireNonNull(reactorNettyResponse);
        this.byteSinkSupplier = () -> {
-           final Sinks.Many<Tuple2<byte[], CompletableFuture<Void>>> outSink = Sinks.many().multicast().onBackpressureBuffer();
-           final Flux<byte[]> byteFlux = outSink.asFlux().map(tup -> {
-                   tup.getT2().complete(null);
-                   return tup.getT1();
-               });
-
-           SinkSubscriber.subscribe(completionSink, Mono.from(reactorNettyResponse.sendByteArray(byteFlux)));
-
+           final Sinks.Many<Tuple2<byte[], CompletableFuture<Void>>> outSink =
+                   Sinks.many().unicast().onBackpressureBuffer();
+           final Flux<Void> flux = outSink.asFlux()
+                   .flatMap(tup ->
+                           Mono.from(reactorNettyResponse.send(
+                                   Mono.just(Unpooled.wrappedBuffer(tup.getT1())),
+                                   bb -> true
+                           )).doOnSuccess(v -> tup.getT2().complete(null))
```

Comment:
You will also have to handle all the other cases, failure/cancellation/etc.

Reply:
I'm thinking we may not need `Sink` stuff if we are going to repeatedly call `NettyOutbound#send.  Seems like we can just move this to `asyncWrite` and have everything write to it.  Maybe we can be hooked into the `close` (or other 'done' hooks) method to complete `completionSink`.